plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java

```diff
@@ -207,6 +235,88 @@ public void testReadAllEntries()
+            throws IOException
+    {
+        MetadataEntry metadataEntry = new MetadataEntry(
+                "metadataId",
+                "metadataName",
+                "metadataDescription",
+                new MetadataEntry.Format(
+                        "metadataFormatProvider",
+                        ImmutableMap.of()),
+                "{\"type\":\"struct\",\"fields\":" +
+                        "[{\"name\":\"ts\",\"type\":\"timestamp\",\"nullable\":true,\"metadata\":{}}]}",
+                ImmutableList.of("part_key"),
+                ImmutableMap.of(),
+                1000);
+        ProtocolEntry protocolEntry = new ProtocolEntry(10, 20, Optional.empty(), Optional.empty());
+        AddFileEntry addFileEntryJsonStats = new AddFileEntry(
+                "addFilePathJson",
+                ImmutableMap.of(),
+                1000,
+                1001,
+                true,
+                Optional.of("{" +
+                        "\"numRecords\":20," +
+                        "\"minValues\":{" +
+                        "\"ts\":\"2960-10-31T01:00:00.000Z\"" +
+                        "}," +
+                        "\"maxValues\":{" +
+                        "\"ts\":\"2960-10-31T02:00:00.000Z\"" +
+                        "}," +
+                        "\"nullCount\":{" +
+                        "\"ts\":1" +
+                        "}}"),
+                Optional.empty(),
+                ImmutableMap.of());
+
+        int numRemoveEntries = 100;
+        Set<RemoveFileEntry> removeEntries = IntStream.range(0, numRemoveEntries).mapToObj(x ->
+                new RemoveFileEntry(
+                        UUID.randomUUID().toString(),
+                        1000,
+                        true))
+                .collect(toImmutableSet());
+
+        CheckpointEntries entries = new CheckpointEntries(
+                metadataEntry,
+                protocolEntry,
+                ImmutableSet.of(),
+                ImmutableSet.of(addFileEntryJsonStats),
+                removeEntries);
+
+        CheckpointWriter writer = new CheckpointWriter(typeManager, checkpointSchemaManager, "test");
+
+        File targetFile = File.createTempFile("testSkipRemoveEntries-", ".checkpoint.parquet");
+        targetFile.deleteOnExit();
+
+        String targetPath = "file://" + targetFile.getAbsolutePath();
+        targetFile.delete(); // file must not exist when writer is called
+        writer.write(
+                entries,
+                createOutputFile(targetPath),
+                ParquetWriterOptions.builder() // approximately 2 rows per row group
+                        .setMaxBlockSize(DataSize.ofBytes(64L))
+                        .setMaxPageSize(DataSize.ofBytes(64L))
+                        .build());
+
+        CheckpointEntryIterator addEntryIterator =
+                createCheckpointEntryIterator(URI.create(targetPath), ImmutableSet.of(ADD), Optional.of(readMetadataEntry(URI.create(targetPath))));
```

Comment:
Very nitpicky, but I'd format this more like

```suggestion
        CheckpointEntryIterator addEntryIterator = createCheckpointEntryIterator(
            URI.create(targetPath),
            ImmutableSet.of(ADD), 
            Optional.of(readMetadataEntry(URI.create(targetPath))));
```